source/jormungandr/tests/departure_board_tests.py

```diff
@@ -1104,3 +1103,107 @@ def test_routes_schedule_with_invalid_forbidden_uri(self):
         _, code = self.query_region("routes/line:A:0/route_schedules?"
                                      "from_datetime=20120615T080000&forbidden_uris[]=toto", check=False)
         assert code == 404
+
+
+@dataset({"timezone_cape_verde_test": {}})
+class TestFirstLastDatetime(AbstractTestFixture):
+    def test_first_last_datetime(self):
+        """
+        test first/last datetime
+
+        * The opening time should be the earliest departure of all vjs of the line
```

Comment:
Those conditions are not mandatory, no? So you're not testing it here, it just in the data provided.
In **data**, we should have `latest_arrival >= opening >= closing >= first_departure` (and ideally your comment is true)